### PR TITLE
Export sd-path functions and beef up systemd-path to show more items

### PR DESCRIPTION
**Pull request by keszybz**
I wanted to export some additional search paths and ended up doing a fairly big cleanup sweep...

### CODE REVIEW

File: man/sd_path_lookup.xml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+  <refsect1>
+    <title>Examples</title>
+
+    <refsect2>
+      <title>Look up the location of ~/Documents</title>
+
+      <programlisting><xi:include href="path-documents.c" parse="text" /></programlisting>
+      <para>Note that the default answer of <filename index='false'>$HOME/Documents</filename> may be
+      overriden by <filename index='false'>user-dirs.conf</filename> or
```
Comment (anitazha) [inline man/sd_path_lookup.xml]: typo: overriden -> overridden

**Comment by keszybz** [comment]
Updated with the typo fixed.

**Comment by keszybz** [comment]
It turns out that I missed the case when XDG_RUNTIME_DIR is not defined. It caused the new test to fail in semaphore. I pushed an additional patch to fix that. LGTM also caught a missing header guard.

**Comment by anitazha** [comment]
bionic CI is feeling better today! Only a timeout on bionic-i386. But now it needs rebase :/

**Comment by keszybz** [comment]
Rebased.

**Comment by poettering** [comment]
btw, the _home vs. _search naming is inherited from the xdg base dir spec: they define env vars ending in _HOME is place to put stuff and then another var for place to search for stuff basically

**Pull request merged**